File: ServicioProductos/SistemaInventarios/SistemaInventario.Dominio/Entidades/Producto.cs
using SistemaInventarios.Dominio.Entidades;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SistemaInventarios.Dominio.Entidades
{
    public class Producto
    {
        public int IdProducto { get; set; }
        public string Nombre { get; set; }
        public string Descripcion { get; set; }
        public string Categoria { get; set; }
        public byte[] Imagen {  get; set; }
        public decimal Precio { get; set; }
        public int Stock { get; set; }
        public DateTime FechaCreacion { get; set; }
        public DateTime FechaModificacion { get; set; }
    }
}

File: FrontEnd/SistemaInventarioFront/sistema-inventario/src/app/services/models/producto.ts
export  class Producto {
    constructor(
      public nombre: string,
      public descripcion: string,
      public categoria: string,
      public imagen: string,
      public precio: number,
      public stock: number,
      public fechaCreacion: Date,
      public fechaModificacion: Date
    ) {
  }
}